File: jEverest/src/ca/marc/everest/datatypes/interfaces/ICodedValue.java
/* 
 * Copyright 2008-2011 Mohawk College of Applied Arts and Technology
 * 
 * Licensed under the Apache License, Version 2.0 (the "License"); you 
 * may not use this file except in compliance with the License. You may 
 * obtain a copy of the License at 
 * 
 * http://www.apache.org/licenses/LICENSE-2.0 
 * 
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
 * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the 
 * License for the specific language governing permissions and limitations under 
 * the License.
 * 
 * User: Justin Fyfe
 * Date: 07-02-2011
 */
package ca.marc.everest.datatypes.interfaces;

import ca.marc.everest.datatypes.generic.*;

/**
 * An interface that represents the structure of a codified value
 */
public interface ICodedValue<T> extends ICodedSimple, IOriginalText {

	/**
	 * Get the code system of the coded simple
	 */
	String getCodeSystem();
	/**
	 * Set the code system of the coded simple
	 */
	void setCodeSystem(String value);
	/**
	 * Get the code system name of the coded simple
	 */
	String getCodeSystemName();
	/**
	 * Set the code system name of the coded simple
	 */
	void setCodeSystemName(String value);
	/**
	 * Get the code system version of the coded simple
	 */
	String getCodeSystemVersion();
	/**
	 * Set the code system version of the coded simple
	 */
	void setCodeSystemVersion(String value);
	/**
	 * Gets the display name of the coded value
	 */
	String getDisplayName();
	/**
	 * Sets the display name of the coded value
	 */
	void setDisplayName(String value);

	/**
	 * Gets the coding rationale for the coded value
	 */
	ISet<CS<CodingRationale>> getCodingRationale();
	/**
	 * Sets the coding rationale for the coded value
	 */
	void setCodingRationale(ISet<CS<CodingRationale>> value);
	
}

File: EV-877/MARC.Everest/DataTypes/Interfaces/ICodedValue.cs
/* 
 * Copyright 2008-2012 Mohawk College of Applied Arts and Technology
 * 
 * Licensed under the Apache License, Version 2.0 (the "License"); you 
 * may not use this file except in compliance with the License. You may 
 * obtain a copy of the License at 
 * 
 * http://www.apache.org/licenses/LICENSE-2.0 
 * 
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
 * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the 
 * License for the specific language governing permissions and limitations under 
 * the License.

 * 
 * User: Justin Fyfe
 * Date: 01-09-2009
 */
using System;
using System.Collections.Generic;
using System.Text;

namespace MARC.Everest.DataTypes.Interfaces
{
    /// <summary>
    /// Represents a coded value
    /// </summary>
    public interface ICodedValue : ICodedSimple, IOriginalText
    {

        /// <summary>
        /// The code system this code mnemonic comes from
        /// </summary>
        string CodeSystem { get; set; }
        /// <summary>
        /// The name of the code system
        /// </summary>
        string CodeSystemName { get; set; }
        /// <summary>
        /// The version of the code system
        /// </summary>
        string CodeSystemVersion { get; set; }
        /// <summary>
        /// Display name for the code
        /// </summary>
        ST DisplayName { get; set; }
        /// <summary>
        /// Identifies the value set that was applicable at the time of coding
        /// </summary>
        string ValueSet { get; set; }
        /// <summary>
        /// Identifies the version of the value set that was applicable at the time of coding
        /// </summary>
        string ValueSetVersion { get; set; }
        /// <summary>
        /// Addtional groups of logically related qualifiers
        /// </summary>
        //[System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Usage", "CA2227:CollectionPropertiesShouldBeReadOnly")]
        //LIST<CDGroup> Group { get; set;  }
        /// <summary>
        /// Coding rationale
        /// </summary>
        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Usage", "CA2227:CollectionPropertiesShouldBeReadOnly")]
        SET<CodingRationale> CodingRationale { get; set;  }
    }
}